pyracf/common/null_response_error.py

```diff
@@ -0,0 +1,45 @@
+"""Exception to use when no data is returned by IRRSMO00."""
+from typing import Union
+
+
+class NullResponseError(Exception):
+    """
+    Raised when no xml string is returned by IRRSMO00.
+    """
+
+    def __init__(self, xml_str: str, run_as_userid: Union[str, None] = None) -> None:
```

Comment:
Will this exception handler print out the return and reason codes for you? I know that they are stored as attributes on the exception handler, but if someone's code doesn't handle this exception and inspect the return and reason codes, the user may never know what happened. This exception should probably always print the return and reason code to facilitate debugging especially in the case where the the return and reason code combination is not one known by pyRACF.

Reply:
I think it would be good for item to be printed on a separate line.

```console

SAF Return Code: 8
RACF Return Code: 200
RACF Reason Code: 16

Check to see if the proper RACF permissions are in place.
For `set` or `alter` functions, you must have at least READ access to `IRR.IRRSMO00.PRECHECK` in the `XFACILIT` class.
```

Reply:
That also makes sense; done!

Reply:
I agree, that looks better. Good call on the spaces, I had missed those. I do think there could be a benefit, but I doubt it would be as useful with those errors. I included them here because it can assist in troubleshooting especially for the "generic" case of a null response due to an unanticipated failure and not the permissions checks we are aware of. With SecurityRequestError, especially in its unified state, the "messages" list really gives the specific context about the failures without the need to directly reference documentation for the meanings of the codes.